sdk/state/state.go

```diff
@@ -161,11 +162,14 @@ func (c *Channel) State() (State, error) {
 	latestCloseSequence := txs.Close.SequenceNumber()
 
 	initiatorEscrowSeqNum := c.initiatorEscrowAccount().SequenceNumber
+	s := c.openAgreement.Envelope.Details.StartingSequence
 
-	if initiatorEscrowSeqNum == c.openAgreement.Envelope.Details.StartingSequence {
+	if initiatorEscrowSeqNum == s {
 		return StateOpen, nil
-	} else if initiatorEscrowSeqNum < latestDeclSequence {
+	} else if initiatorEscrowSeqNum < latestDeclSequence && initiatorEscrowSeqNum%2 == s%2 {
 		return StateClosingWithOutdatedState, nil
+	} else if initiatorEscrowSeqNum < latestDeclSequence && initiatorEscrowSeqNum%2 != s%2 {
```

Comment:
💡 I think these operations would fit well as utility functions in the `sdk/txbuild` package.

I'm thinking:

```go
type TransactionType string

const (
	TransactionTypeDeclaration TransactionType = "declaration"
	TransactionTypeClose       TransactionType = "close"
)

func SequenceNumberToTransactionType(startingSeqNum, seqNum int64) TransactionType {
	// ...
}
```

Potentially we could do a small refactor on where we set the sequence numbers in the in the other txbuild functions, to call something like:
```go
func SequenceNumberFromTransactionType(startingSeqNum int64, tt TransactionType) int64  {
	// ...
}
```

Reply:
I like it https://github.com/stellar/experimental-payment-channels/pull/324/commits/cbcd5416e68d803cbf08ab73d76c4e8b5fcd5d49

second suggestion sgtm, I can do in a separate PR